nym-api/src/node_status_api/handlers.rs

```diff
@@ -0,0 +1,585 @@
+// Copyright 2021-2024 - Nym Technologies SA <contact@nymtech.net>
+// SPDX-License-Identifier: GPL-3.0-only
+
+use crate::node_status_api::helpers::{
+    _compute_mixnode_reward_estimation, _gateway_core_status_count, _gateway_report,
+    _gateway_uptime_history, _get_active_set_detailed, _get_gateway_avg_uptime,
+    _get_gateways_detailed, _get_gateways_detailed_unfiltered, _get_mixnode_avg_uptime,
+    _get_mixnode_inclusion_probabilities, _get_mixnode_inclusion_probability,
+    _get_mixnode_reward_estimation, _get_mixnode_stake_saturation, _get_mixnode_status,
+    _get_mixnodes_detailed, _get_mixnodes_detailed_unfiltered, _get_rewarded_set_detailed,
+    _mixnode_core_status_count, _mixnode_report, _mixnode_uptime_history,
+};
+use crate::node_status_api::models::AxumResult;
+use crate::v2::AxumAppState;
+use axum::extract::{Path, Query, State};
+use axum::Json;
+use axum::Router;
+use nym_api_requests::models::{
+    AllInclusionProbabilitiesResponse, ComputeRewardEstParam, GatewayBondAnnotated,
+    GatewayCoreStatusResponse, GatewayStatusReportResponse, GatewayUptimeHistoryResponse,
+    GatewayUptimeResponse, InclusionProbabilityResponse, MixNodeBondAnnotated,
+    MixnodeCoreStatusResponse, MixnodeStatusReportResponse, MixnodeStatusResponse,
+    MixnodeUptimeHistoryResponse, RewardEstimationResponse, StakeSaturationResponse,
+    UptimeResponse,
+};
+use nym_mixnet_contract_common::MixId;
+
+pub(crate) fn node_status_routes(network_monitor: bool) -> Router<AxumAppState> {
+    // in the minimal variant we would not have access to endpoints relying on existence
```

Comment:
again, if it would make your life and routing easier, perhaps always expose **all** the routes, but simply immediately return an error in the handler if one of the conditions is not fulfilled, such as in this case network monitor not being enabled

Reply:
axum routes don't have to behave exactly 100% identical to what we have in rocket

Reply:
> always expose all the routes, but simply immediately return an error in the handler if one of the conditions is not fulfilled

Doesn't it make much more sense to not expose routes which shouldn't be there and not return unnecessary 4xx/5xx ?

 It's better from both code and API design perspective